common/cache_managers.py

```
@@ -0,0 +1,565 @@
+
+                if data is not None:
+                    # Cache in both levels
+                    self._module_cache[cache_key] = data
+                    cache.set(full_cache_key, data, timeout=self.cache_timeout)
+
+                    db_duration = time.time() - db_start_time
+                    logger.info(
+                        f"Successfully cached {self.cache_prefix}:{cache_key} in {db_duration:.2f}s"
+                    )
+                    return data
+                else:
```

Ruff (0.12.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)